integration-tests/utils/integration-utils.js

```diff
@@ -181,24 +182,53 @@ const delay = ms => {
 }
 
 const dismissNotifications = async page => {
-  await delay(3000) // delayed for any alerts to show
-  const notificationCloseButtons = await page.$x(
-    '(//a | //*[@role="button"] | //button)[contains(@aria-label,"Dismiss") and not(contains(@aria-label,"error"))]')
+  const notificationCloseButtonXpath = '(//a | //*[@role="button"] | //button)[contains(@aria-label,"Dismiss") and not(contains(@aria-label,"error"))]'
+  await Promise.race([
+    delay(3000), // delayed for any alerts to show
+    page.waitForXPath(notificationCloseButtonXpath, { visible: true })
+  ])
 
+  const notificationCloseButtons = await page.$x(notificationCloseButtonXpath)
   await Promise.all(
-    notificationCloseButtons.map(handle => handle.click())
+    notificationCloseButtons.map(button => button.evaluateHandle(btn => btn.click(), button))
   )
+  return !!notificationCloseButtons.length && delay(1000)
+}
 
-  return !!notificationCloseButtons.length && delay(1000) // delayed for alerts to animate off
+const dismissNPSSurvey = async page => {
+  let element
+  try {
+    element = await page.waitForXPath('//iframe[@aria-label="NPS Survey"]', { timeout: 1000 })
+  } catch (e) {
+    // NPS survey was not found
+    return
+  }
+  try {
+    console.log('dismissing NPS survey')
+    const iframe = await element.contentFrame()
+    const [closeButton] = await iframe.$x('.//*[@id="ask-me-later"]')
+    await closeButton.evaluateHandle(button => button.click(), closeButton)
+    await delay(500) // delayed for survey to animate off
+  } catch (e) {
+    console.error(e)
+    throw e
+  }
 }
 
 const signIntoTerra = async (page, { token, testUrl }) => {
-  !!testUrl && await page.goto(testUrl, navOptionNetworkIdle())
-  await page.waitForXPath('//*[contains(normalize-space(.),"Loading Terra")]', { hidden: true, timeout: 60 * 1000 })
+  if (!!testUrl) {
+    await page.goto(testUrl, navOptionNetworkIdle())
+    // Wait for id="root" element first to prevent checking Loading Terra... spinner too soon
+    await page.waitForXPath('//*[@id="root"]')
+    await page.waitForXPath('//*[contains(normalize-space(text()),"Loading Terra")][./*[contains(@src, "/loading-spinner.svg")]]', { hidden: true })
+  }
+
   await waitForNoSpinners(page)
-  await page.waitForFunction('!!window["forceSignIn"]')
+  await page.waitForFunction('window["forceSignIn"]', { polling: 100 })
   await page.evaluate(token => window.forceSignIn(token), token)
+
   await dismissNotifications(page)
+  await dismissNPSSurvey(page)
```

Comment:
NPS Survey affected three UI tests: `analysis-context-bar`, `run-analysis` and `run-notebook` because these tests start with registering (creating) a new user.
When NPS Survey is displayed (randomly), it blocks clicking "View Workspaces" link.
<img width="1142" alt="Screen Shot 2022-06-27 at 7 20 26 PM" src="https://user-images.githubusercontent.com/35533885/176089512-422f656d-712b-48b4-9c6b-d868251f54b4.png">

Reply:
thanks for finding this!